trestle/core/commands/author/profile.py

```diff
@@ -336,47 +348,38 @@ def assemble_profile(
                     for part in add.parts:
                         if part.name not in allowed_sections:
                             raise TrestleError(f'Profile has alter with name {part.name} not in allowed sections.')
-        ProfileAssemble._replace_alter_adds(new_profile, found_alters)
+        ProfileAssemble._replace_alter_adds(parent_prof, found_alters)
         if set_parameters:
-            ProfileAssemble._replace_modify_set_params(new_profile, param_dict, param_map)
-
-        new_profile_dir = trestle_root / f'profiles/{new_profile_name}'
-        new_profile_path = new_profile_dir / 'profile.json'
+            ProfileAssemble._replace_modify_set_params(parent_prof, param_dict, param_map)
 
         if version:
-            new_profile.metadata.version = com.Version(__root__=version)
-
-        if new_profile_path.exists():
-            # first do simple and fast tests of changes so that the full equality check is only done as last resort
-            _, _, existing_profile = ModelUtils.load_distributed(new_profile_path, trestle_root)
-            # need special handling in case modify is None in either or both profiles
-            no_changes = not existing_profile.modify and not new_profile.modify
-            both_modifies_present = existing_profile.modify and new_profile.modify
-            if both_modifies_present:
-                no_changes = (
-                    existing_profile.modify.alters == new_profile.modify.alters
-                    and existing_profile.modify.set_parameters == new_profile.modify.set_parameters
-                )
-            if no_changes and existing_profile.metadata.version != new_profile.metadata.version:
-                no_changes = False
-            # at this point if no known changes then do the full and possibly expensive comparison to confirm
-            if no_changes and existing_profile == new_profile:
+            parent_prof.metadata.version = com.Version(__root__=version)
+
+        assem_prof_path = ModelUtils.path_for_top_level_model(
+            trestle_root, assem_prof_name, prof.Profile, new_content_type
+        )
+
+        if assem_prof_path.exists():
```

Comment:
Better to check for non existence and raise exception if the case

Reply:
This is ok - it's just loading the file if it exists.  It doesn't need to exist.